src/schemas/ProgramDonorSummary/resolvers/summaryStats.ts

```diff
@@ -0,0 +1,143 @@
+//@ts-ignore no type defs
+import stringify from 'json-stringify-deterministic';
+import esb from 'elastic-builder';
+
+import { GlobalGqlContext } from 'app';
+import { GraphQLFieldResolver } from 'graphql';
+import {
+  ProgramDonorSummaryStats,
+  ProgramDonorSummaryFilter,
+  ElasticsearchDonorDocument,
+  ProgramDonorSummaryStatsGqlResponse,
+  EsDonorDocumentField,
+} from './types';
+import { Client } from '@elastic/elasticsearch';
+import { ELASTICSEARCH_PROGRAM_DONOR_DASHBOARD_INDEX } from 'config';
+
+const programDonorSummaryStatsResolver: (
+  esClient: Client,
+) => GraphQLFieldResolver<
+  unknown,
+  GlobalGqlContext,
+  {
+    programShortName: string;
+    filters: ProgramDonorSummaryFilter[];
+  }
+> = esClient => async (source, args, context): Promise<ProgramDonorSummaryStatsGqlResponse> => {
+  const { programShortName, filters } = args;
+
+  type AggregationName = keyof ProgramDonorSummaryStats | 'donorsWithAllCoreClinicalData';
+
+  const filterAggregation = (name: AggregationName, filterQuery?: esb.Query | undefined) =>
+    esb.filterAggregation(name, filterQuery);
+
+  const esQuery = esb
+    .requestBodySearch()
+    .query(esb.boolQuery().must([esb.matchQuery(EsDonorDocumentField.programId, programShortName)]))
+    .aggs([
+      filterAggregation('fullyReleasedDonorsCount').filter(
+        esb
+          .termsQuery()
+          .field(EsDonorDocumentField.releaseStatus)
+          .values(['FULLY_RELEASED'] as ElasticsearchDonorDocument['releaseStatus'][]),
+      ),
+      filterAggregation('partiallyReleasedDonorsCount').filter(
+        esb
+          .termsQuery()
+          .field(EsDonorDocumentField.releaseStatus)
+          .values(['PARTIALLY_RELEASED'] as ElasticsearchDonorDocument['releaseStatus'][]),
+      ),
+      filterAggregation('noReleaseDonorsCount').filter(
+        esb
+          .termsQuery()
+          .field(EsDonorDocumentField.releaseStatus)
+          .values(['NO_RELEASE', ''] as ElasticsearchDonorDocument['releaseStatus'][]),
+      ),
+      filterAggregation('donorsProcessingMolecularDataCount').filter(
+        esb
+          .termsQuery()
+          .field(EsDonorDocumentField.processingStatus)
+          .values(['PROCESSING'] as ElasticsearchDonorDocument['processingStatus'][]),
+      ),
+      filterAggregation('donorsWithReleasedFilesCount').filter(
+        esb
+          .termsQuery()
+          .field(EsDonorDocumentField.processingStatus)
+          .values(['COMPLETE'] as ElasticsearchDonorDocument['processingStatus'][]),
```

Comment:
Shouldn't it be an aggregate of donors with `relaseStatus` === `COMPLETE` or `PARTIAL`?? Or am I misunderstanding this field.